include/llvm/CodeGen/TargetLowering.h

```diff
@@ -3358,6 +3358,15 @@ class TargetLowering : public TargetLoweringBase {
     return false;
   }
 
+  /// For some targets, an LLVM type must be broken down into multiple
+  /// smaller types. Usually the halves are ordered according to the endianness
+  /// but for some platform that would break. So this method will default to
+  /// matching the endianness but can be overridden.
+  virtual bool
+  functionArgumentsSplitLittleEndian(const DataLayout &DL) const {
```

Comment:
Nit: I recommend giving this a slightly clearer name. For example, many of the boolean methods in this class start with `should` or `is` - perhaps `shouldSplitFunctionArgumentsAsLittleEndian`

Reply:
ok, I was copying the pattern from the existing function above but I'll happily switch to your suggestion :)